src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -528,6 +529,20 @@ public T WithClientCapabilities(IEnumerable<string> clientCapabilities)
             return (T)this;
         }
 
+        /// <summary>
+        /// Determines whether or not instance discovery is performed when attempting to authenticate. Setting this to false will completely disable
+        /// instance discovery and authority validation. This will not affect the behavior of application configured with regional endpoints however.
+        /// Also, you can still provide your own instance metadata.
+        /// </summary>
+        /// <param name="enableInstanceDiscovery">Determines if instance discovery/Authority validation is performed</param>
+        /// <returns></returns>
+        public T WithInstanceDiscoevryEndpoint(bool enableInstanceDiscovery)
+        {
+            Config.InstanceDiscoveryEndpoint = Constants.InstanceDiscoveryEndpointDisabled;
```

Comment:
Why is `InstanceDiscoveryEndpoint` a string and not a bool? It should be called `IsInstanceDiscoveryEnabled` as per suggestion of naming bool fields by @SameerK-MSFT